enhancements/olm/runtime-bundle-configuration.md

```diff
@@ -0,0 +1,858 @@
+
+## Operational Aspects of API Extensions
+
+### Impact of Install Mode Extensions
+
+**ClusterExtension API Enhancement:**
+- **Architectural Impact:** The `config.inline.watchNamespace` field enables runtime install mode selection, moving from compile-time (bundle) to runtime (installation) configuration
+- **Operational Impact:**
+    - Administrators must understand namespace relationships and RBAC implications
+    - Troubleshooting requires awareness of install vs watch namespace distinctions
+    - Monitoring and alerting must account for namespace-scoped operator deployments
+
+**RBAC Resource Generation:**
+- **Architectural Impact:** Dynamic RBAC generation based on install mode creates different permission patterns for the same operator
+- **Operational Impact:**
+    - Permission debugging requires understanding of install mode impact on RBAC scope
+    - Security auditing must consider namespace-level vs cluster-level permission grants
+
+### Impact on Existing SLIs
+
+With the removal of the install mode concept in OLMv1, operator packages that want to continue to use this stop-gap feature are expected to surface configuration documentation, and call out if the `watchNamespace` parameter is a part of it, along with usage examples etc. Until a broad percentage of operator packages, especially those that don't support `AllNamespaces` mode, take action to make such documentation available, a spike in bad installations is expected.
+
+**Installation Success Rate:**
+
+*   **RBAC Validation Complexity:** Namespace-scoped installations require more complex RBAC validation to ensure the ServiceAccount has appropriate permissions for the target namespace. RBAC misconfigurations that work in AllNamespaces mode may fail in Single/OwnNamespace modes.
+    *   Example: ServiceAccount has cluster-wide read permissions but lacks namespace-specific write permissions, causing installation to fail.
+*   **Bundle Compatibility Validation:** Additional validation layer to confirm bundles support the requested install mode. Bundles that only support AllNamespaces will fail when Single/OwnNamespace is requested.
+    *   Example: Attempting to install a bundle with `watchNamespace: "test"` when the bundle CSV only declares support for AllNamespaces install mode.
+
+**Installation Time:**
+
+*   **RBAC Generation Complexity:** Converting CSV `permission` to namespace-scoped Role/RoleBinding resources requires additional processing time. Complex operators with extensive permission requirements will see increased installation duration.
+
+**Operator Availability:**
+
+*   **ServiceAccount Permission Dependencies:** Namespace-scoped operators depend on ServiceAccount permissions that may be modified by namespace administrators, creating additional failure points not present in cluster-scoped installations.
+    *   Example: Namespace admin removes critical RoleBinding, causing operator to lose access to required resources.
```

Comment:
Wouldn't OLMv1 re-create this `RoleBinding`?

Reply:
Also, how is this different than a cluster admin deleting a ClusterRoleBinding when an operator is watching all namespaces?